dadjokes-onchain/next/src/hooks/useVote.js

```diff
@@ -0,0 +1,40 @@
+import { sepolia } from "viem/chains";
+import { parseEther } from "viem/utils";
+
+export function useVote(dadJokesContract, walletClient, publicClient) {
+  async function handleVote(index, type) {
+    const reward = type + 1;
+    // Instantiate a Wallet Client and a Public Client
+    // const walletClient = await ConnectWalletClient();
+    // Retrieve the wallet address using the Wallet Client
+    const [address] = await walletClient.requestAddresses();
+    await walletClient.switchChain({ id: sepolia.id });
+
+    let rewardAmount;
+
+    switch (type) {
+      case 0:
+        rewardAmount = parseEther("0.001");
+        break;
+      case 1:
+        rewardAmount = parseEther("0.005");
+        break;
+      case 2:
+        rewardAmount = parseEther("0.01");
+        break;
+      default:
+        throw new Error("Invalid reward type");
+    }
+    const { request } = await publicClient.simulateContract({
```

Comment:
is it worth explaining what simulateContract does in comments? the viem js docs on this is not as beginner friendly..and the "simulate" verb can be confusing to newbies.